cpp/ql/lib/change-notes/2022-11-21-ir-guards-replacement.md

```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* The `Guards` library has been replaced with the API-compatible `IRGuards` implementation, which has better precision in some cases.
```

Comment:
Which of the test cases should I look at to get an idea of where `IRGuards` might have better precision?

Reply:
The GuardsControl test is probably the most relevant. Note that there's currently still one issue, which should be fixed by https://github.com/github/codeql/pull/15021

Reply:
One clear example I'm aware of is the ternary operator: the old `Guards` library doesn't handle that operation very well, whereas the new `IRGuards` handles that just fine.